File: dayOne/JavaScript/LevelTwo/LevelTwoFunctions.js
function getRepeatedValues(numbers) {

    let sizeCount = 0;
    
    for (let index = 0; index < numbers.length; index++) {
        for (let indexCheck = index + 1; indexCheck < numbers.length; indexCheck++) {
            if (numbers[index] === numbers[indexCheck]) {
                sizeCount++;
            }
        }
    }
    
    
    let repeatedValues = new Array(sizeCount).fill(0);
    let resultIndex = 0;
    
    for (let arrayIndex = 0; arrayIndex < numbers.length; arrayIndex++) {
        for (let indexCheck = arrayIndex + 1; indexCheck < numbers.length; indexCheck++) {
            if (numbers[arrayIndex] === numbers[indexCheck]) {
                repeatedValues[resultIndex] = numbers[arrayIndex];
                resultIndex++;
            }
        }
    }
    return repeatedValues;
}

let numbers = [1, 2, 3, 2, 4, 3];
let newArray = getRepeatedValues(numbers);

console.log(numbers);
console.log(newArray);

console.log();



function moveAllZeroes(numbers) {
    let tempArray = new Array(array.length).fill(0);
    let newIndex = 0;
        
    for (let index = 0; index < array.length; index++) {
        if (array[index] != 0) {
           tempArray[newIndex] = array[index];
           newIndex++;
        }          
    }
  
    return tempArray;
}

let array = [5, 0, 3, 0, 2, 0];
let tempArray = moveAllZeroes(array);

console.log(array);
console.log(tempArray);

File: dayOne/Java/LevelTwo/LevelTwoFunctions.java
import java.util.Arrays;

public class LevelTwoFunctions {

    public static void main(String[] args) {
    
        int[] numbers = {1, 2, 3, 2, 4, 3};
        
        int[] newArray =  getRepeatedValues(numbers);
        
        System.out.println(Arrays.toString(numbers));        
        System.out.println(Arrays.toString(newArray));  
    
    }
    
    
    public static int[] getRepeatedValues(int[] array) {
    
        int sizeCount = 0;
        
        for (int index = 0; index < array.length; index++) {
            for (int indexCheck = index + 1; indexCheck < array.length; indexCheck++) {
                if (array[index] == array[indexCheck]) {
                    sizeCount++;
                }
            }
            
        }
        
        int[] repeatedValues = new int[sizeCount];
        int index = 0;
        
        for (int arrayIndex = 0; arrayIndex < array.length; arrayIndex++) {
            for (int indexCheck = arrayIndex + 1; indexCheck < array.length; indexCheck++) {
                if (array[arrayIndex] == array[indexCheck]) {
                    repeatedValues[index] = array[arrayIndex];
                    index++;
                }
            }
        }   
        return repeatedValues;
    }
    
}
